### DISCUSSION TITLE
Symbolic State satisfies double should never be called

### DISCUSSION THREAD

**Discussion by Szpilman2** [UPPAALModelChecker/UPPAAL-Meta#262] [Q&A] [answered]
Dear UPPAAL team,

I hope this message finds you well.

I recently came across your paper titled [An Introduction to Schedulability Analysis Using Timed Automata](https://homes.cs.aau.dk/~kgl/China2011/Slides/QuasimodoHandbook.pdf) (page 108) and noticed that in the paper, you have treated the WCET (worst case execution time) as a natural number.
![wcet](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/35931007/631bc884-f7dd-407c-86f9-9947da802b3f)

However, I was under the impression that WCET is a double. When I attempted to create a model with "double" clock bounds, UPPAAL returned an error message like this:
![Error](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/35931007/122cd2ac-e353-4f06-9ef8-69291a3cbaa2)

also my simple model is like this:
![Model](https://github.com/UPPAALModelChecker/UPPAAL-Meta/assets/35931007/99fea82f-79a1-42a2-8b44-63bcebddaabf)
(in this case 1.5 (ms) is WCET)

could you possibly help me that how can I use double clock bounds in my guards? (or  I may be mistaken in assuming that WCET is a double?)

thank you so much.
best regards.

**Comment by mikucionisaau** [comment, marked as answer]
The floating point types are supported only in concrete state engine and the symbolic (triggered by query `E<>`) does not support floating point types due to their dense value domain.

As you can see the paper refers to calligraphic `N` -- set of natural numbers, i.e. integers, and that's what timed automata assume.

Floating point numbers were added later to support performance analysis, where arbitrary operations a available and are beyond reachability decidability.

If you need to support fractions, then you can find a common denominator of all the constants and scale all of them to produce integer values everywhere.

**Reply by Szpilman2** [reply]
Thank you very much for your clear and excellent explanation.

Best regards
